### ISSUE TITLE
Support deploying JARs and POMs to Maven compatible repositories

### ISSUE THREAD

**Issue by aalmiray** [jreleaser/jreleaser#953]
Sonatype Nexus2 powers Maven Central for the time being. Publishing artifacts to Maven Central has been requested since the early days.

JReleaser may need a new wrokflow step: `deploy`.
JARs & POMs should be "staged" using the build tool of choice that is, files should be placed at a local directory using the default layout expected by Nexus. For example

```
target/
└── staging-deploy
    └── test
        └── jreleaser
            └── app
                ├── 1.0.0
                │   ├── app-1.0.0.pom
                │   ├── app-1.0.0.pom.asc
                │   ├── app-1.0.0.pom.asc.md5
                │   ├── app-1.0.0.pom.asc.sha1
                │   ├── app-1.0.0.pom.md5
                │   └── app-1.0.0.pom.sha1
                ├── maven-metadata.xml
                ├── maven-metadata.xml.md5
                └── maven-metadata.xml.sha1
```

Additionally, JReleaser may perform the following tasks before deploying takes place:

- pom files comply with the minimum set of elements (via [pomchecker](https://kordamp.org/pomchecker/pomchecker-cli/index.html))
- `-sources` & `-javadoc` JARs are available
- generate `sha256` and `sha512` checksums
- sign all artifacts when PGP is configured. Avoids configuring PGP on the build tool

**Comment by hboutemy** [comment]
notice that `.asc.md5` and `.asc.sha1` and pther checksums for `.asc` are not useful (even if they are not forbidden)

**Comment by aalmiray** [comment]
What do you mean with "not useful"? AFAICT there's a checksum rule that triggers when there are no matching checksum files.

**Comment by hboutemy** [comment]
no, not for `.asc` because they are not an attached artifact
see for example https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-artifact-plugin/3.3.0/ and yo9uo'll see that `.asc` is there, but no checksums for it

**Comment by aalmiray** [comment]
Oh, oh, So it's the _gradle_ plugin I'm using the one that calculates checksums for signature files ... well what a surprise there 😏 

Thus, I only have to calculate checksums for JARs & POMs, but not for signatures, right?

**Comment by aalmiray** [comment]
Also support:

- [ ] JFrog Artifactory
- [ ] Gitea Packages
- [ ] Gitlab Packages
- [ ] Github packages

**Comment by aalmiray** [comment]
Released in v1.3.0 -> https://github.com/jreleaser/jreleaser/releases/tag/v1.3.0
